src/inmanta/export.py

```diff
@@ -360,15 +363,16 @@ def run(
         resource_sets_to_remove: Optional[Sequence[str]] = None,
     ) -> Union[Tuple[int, ResourceDict], Tuple[int, ResourceDict, Dict[str, ResourceState], Optional[Dict[str, Any]]]]:
         """
-        Run the export functions
+        Run the export functions. Return value for partial json export uses 0 as version placeholder.
         """
         if not partial_compile and resource_sets_to_remove:
             raise Exception("Cannot remove resource sets when a full compile was done")
         resource_sets_to_remove_all: List[str] = list(resource_sets_to_remove) if resource_sets_to_remove is not None else []
 
         self.types = types
         self.scopes = scopes
-        self._version = self.get_version(no_commit)
+
+        self._version = self.get_version(no_commit) if not partial_compile else None
```

Comment:
Mostly because 0 is only valid under the assumption that it is not a valid version number for a full compile. That is indeed the case but since that is an implementation detail I thought it would be better not to rely on it and to use `None` instead to declare that there is no version associated with this export at this point. If you do prefer 0 though I have no actual objection to changing it.

Reply:
I would go for 0